projects/osquery/build.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -eu
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+PROJECT=osquery
+
+# Move the project content into the current overlay.
+# CMake builtin 'rename' will attempt a hardlink.
+( cd / &&\
+  mv "${SRC}/${PROJECT}" "${SRC}/${PROJECT}-dev" &&\
+  mv "${SRC}/${PROJECT}-dev" "${SRC}/${PROJECT}" )
+
+pushd "${SRC}/${PROJECT}"
+mkdir build && pushd build
+
+export CXXFLAGS="${CXXFLAGS} -Wl,-lunwind -Wl,-lc++abi"
```

Comment:
It's just a coincidence that OSS-Fuzz builder has unwind library installed (we meant to delete it, but somehow forgot, filed https://github.com/google/oss-fuzz/issues/2947 for now). This library is not present in the fuzzing images and therefore fuzz targets should not rely on its presence. Preferably, the library should be statically linked in fuzz targets binaries. Alternatively, it needs to be put in the output directory and the proper `RPATH` needs to be set up for fuzz target binaries (to make sure they can discover the library using a relative path).

Proxygen project had that issue too, you can you copy over the following snippets (from https://github.com/google/oss-fuzz/blob/d488ec15e7548f0bf5c73823782fb2fb2794d210/projects/proxygen/build.sh#L20):

```
# Link to, and copy over, libunwind
# We need to link to libunwind to ensure exception handling works
# See https://clang.llvm.org/docs/Toolchain.html#unwind-library
export LDFLAGS="-lunwind"
mkdir -p $OUT/lib
cp /usr/lib/x86_64-linux-gnu/libunwind.so.8 $OUT/lib/
```

and from https://github.com/google/oss-fuzz/blob/master/projects/proxygen/build.sh#L30:

```
find ./_build/proxygen/fuzzers -type f -executable -exec patchelf --set-rpath '$ORIGIN/lib' {} \;
```

Reply:
We should not need unwind directly but dependency configure scripts may check/detect/use it since it is available. 

The best fix is indeed to remove it from the system if it is not intended to be available. I will confirm if unwind is linked statically or dynamically.

Reply:
Ah, if it's not needed during the runtime, then it's fine!